test_agent/rust/src/testagent.rs

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use async_trait::async_trait;
+use serde_json::Value;
+use up_rust::{Data, UCode, UListener};
+use up_rust::{UMessage, UStatus, UTransport};
+
+use std::io::{Read, Write};
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+use serde::Serialize;
+
+use crate::utils::{convert_json_to_jsonstring, WrapperUMessage, WrapperUUri};
+use crate::*;
+
+#[derive(Serialize)]
+pub struct JsonResponseData {
+    data: HashMap<String, String>,
+    action: String,
+    ue: String,
+}
+
+pub struct SocketTestAgent {
+    utransport: UtransportSocket,
+    clientsocket: Arc<Mutex<TcpStreamSync>>,
+    clientsocket_to_tm: Arc<Mutex<TcpStreamSync>>,
+    listner_map: Vec<String>,
```

Comment:
Is this a hold-over from when `register_listener()` returned a String we would use to call `unregister_listener()`?